### PR TITLE
Initial generation of AWS patches

### PR DESCRIPTION
**Pull request by mattBrzezinski**
## Overview
Testing on AWS can be tricky, at times you might not want to actually generate resources. This package is here to mock things for you! It runs off similar principles as [AWS.jl](https://github.com/JuliaCloud/AWS.jl), by taking advantage of the handy API definitions provided by the [aws/aws-sdk-js](https://github.com/aws/aws-sdk-js/tree/master/apis) repository.

This package works by generating Julia functions based off of `example.json` files simulating responses back from various AWS services. Using the `@patches` macro it then is able to attach these patches to existing service modules defined by the `AWS.jl` package.

An example taken from the README:
```julia
using AWS
using MockAWS
using Mocking

Mocking.activate()

@service S3
@patches S3

apply(S3.get_object_1_patch) do
  @mock S3.get_object("bucket", "key")
end

Dict{String,Any} with 8 entries:
  "ETag"          => "\"6805f2cfc46c0f04559748bb039d69ae\""
  "LastModified"  => "Thu, 15 Dec 2016 01:19:41 GMT"
  "ContentLength" => "3191"
  "Metadata"      => Dict{String,Any}()
  "AcceptRanges"  => "bytes"
  "VersionId"     => "null"
  "ContentType"   => "image/jpeg"
  "TagCount"      => 2
```

## Notes
- Not all AWS services have example outputs defined for them, only what AWS provides us with do we provide patches
- There maybe multiple examples for a services endpoint, which is why patches have a number tacked in their names
- When attaching patches with the `@patch` macro, the Symbol passed in **must** be the same as that used to create the high-level `AWS.jl` module

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,40 @@
...
 Mocking Framework for AWS.jl
+
+## Overview
+This package generates patch functions for all AWS services which define them in the [aws-sdk-js](https://github.com/aws/aws-sdk-js/tree/master/apis) repository.
+It is meant to provide means of testing AWS service functionality without needing to actually generate resources.
+
+## Usages
+Patches are only available to [high-level](https://github.com/JuliaCloud/AWS.jl/tree/master/src/services) AWS.jl definitions, and are limited to only services which AWS themselves have defined example outputs.
+
+To use this package you must first must define the high-level AWS service, then attach the patches to the module.
```
Comment (arnaudh) [inline README.md]:
```suggestion
To use this package you must first define the high-level AWS service, then attach the patches to the module.
```

File: README.md [hunk 2]
```diff
@@ -1,2 +1,40 @@
 # mAWS.jl
 Mocking Framework for AWS.jl
+
+## Overview
+This package generates patch functions for all AWS services which define them in the [aws-sdk-js](https://github.com/aws/aws-sdk-js/tree/master/apis) repository.
```
Comment (arnaudh) [inline README.md:5]: Maybe add/link instructions on how to generate the code in `src/patches`, under a `## Contributing` section?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,40 @@
 # mAWS.jl
```
Comment (arnaudh) [inline README.md]: Did you consider the more explicit name `MockAWS.jl`? This way people would know what this package does just based on the name, without having to read the description.

Comment (arnaudh) [reply README.md]: Cool package by the way!

File: README.md (hunk 2 above)
Comment (mattBrzezinski) [reply README.md:5]: The `.github/workflows/update_patches.yml` runs every night to regenerate and create a merge request if patches are added.

But will this in regardless!

File: README.md (hunk 3 above)
Comment (mattBrzezinski) [reply README.md]: Just changed it :)

File: README.md [hunk 4]
```diff
@@ -1,2 +1,48 @@
...
+
+## Overview
+This package generates patch functions for all AWS services which define them in the [aws-sdk-js](https://github.com/aws/aws-sdk-js/tree/master/apis) repository.
+It is meant to provide means of testing AWS service functionality without needing to actually generate resources.
+
+## Usages
+Patches are only available to [high-level](https://github.com/JuliaCloud/AWS.jl/tree/master/src/services) AWS.jl definitions, and are limited to only services which AWS themselves have defined example outputs.
+
+To use this package you must first define the high-level AWS service, then attach the patches to the module.
+**NOTE:** When applying patches, you must use the same name as you used when defining the AWS.jl high-level service module.
```
Comment (mjram0s) [inline README.md]:
```suggestion
**NOTE:** When applying patches, you must use the same name you used when defining the AWS.jl high-level service module.
```

File: src/MockAWS.jl [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+using JSON
+using Mocking
+using OrderedCollections: OrderedDict
+
+export @patches, generate_patches
+
+"""
+    macro patches(service::Symbol)
+
+Attach all patches to a high-level AWS.jl service.
```
Comment (mjram0s) [inline src/MockAWS.jl]:
```suggestion
Attach all patches to a high-level AWS.jl `service`.
```

**Comment by mjram0s** [comment]
LGTM

**Comment by mjram0s** [comment]
I think you should add one test per service

**Comment by mattBrzezinski** [comment]
> I think you should add one test per service

They all return back dictionaries which sometimes are empty, not sure if there's a value in adding in a test as they all work in the same way.

**Pull request merged**